core/src/main/java/org/elasticsearch/index/query/GeoShapeQueryBuilder.java

```diff
@@ -237,12 +236,11 @@ public ShapeRelation relation() {
 
     @Override
     protected Query doToQuery(QueryShardContext context) throws IOException {
```

Comment:
if we don't do any parsing as part of toQuery, I think the throws clause can go away from its signature everywhere, same for doToQuery